### PR TITLE
Introduce `TestStorageHelper` for storage boilerplate actions

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

When removing/refactoring re-upload task repeated actions cam into play so extracting this into separate helper class. We can use this in all other tests so to reduce the clutter of all this setup

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    public static TestStorageHelper getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new TestStorageHelper();
+        }
+
+        return INSTANCE;
+    }
+
+    public static void createDocker() {
+        dockerContainer = new DockerComposeContainer<>(
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:40]: why not use Class rule
```
@ClassRule
public static DockerComposeContainer environment =
    new DockerComposeContainer(new File("src/test/resources/compose-test.yml"))
            .withExposedService("redis_1", REDIS_PORT)
            .withExposedService("elasticsearch_1", ELASTICSEARCH_PORT);

```

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        } catch (InvalidKeyException | URISyntaxException exception) {
+            throw new RuntimeException("Unable to setup docker environment", exception);
+        }
+    }
+
+    public static void stopDocker() {
+        dockerContainer.stop();
+    }
+
+    public void createContainer() {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:68]: let's specifically call this method createBulkscanContainer ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java (hunk 1 above)
Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:40]: `ClassRule` is junit4 and we should finally make progress to junit5 instead ;). it's tricky as it needs a lot of other reviews to be done beforehand ;)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            throw new RuntimeException(
+                "Cannot start container. Did you create one with "
+                    + TestStorageHelper.class.getSimpleName()
+                    + "::createDocker in a static context?"
+            );
+        }
+
+        try {
+            dockerContainer.start();
+            cloudBlobClient = CloudStorageAccount
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:56]: method says start docker but doing more than just starting :)

Comment (nitinprabhu) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:56]: this should be outside or method should be something else

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import static uk.gov.hmcts.reform.bulkscanprocessor.helper.DirectoryZipper.zipDir;
+
+public class TestStorageHelper {
+
+    private static TestStorageHelper INSTANCE;
+
+    public static final String CONTAINER_NAME = "bulkscan";
+    public static final String ZIP_FILE_NAME = "1_24-06-2018-00-00-00.zip";
+
+    private static DockerComposeContainer<?> dockerContainer;
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:23]: do we really need generics here ? can be just `DockerComposeContainer`

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:23]: Just solving the warning here ;). Class is generic

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java (hunk 3 above)
Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:56]: Well. It is "StorageHelper" afterall. Could call `setUp` but it's just too generic I guess. I don't mind what it is now but if you have something specific in mind - let me know ;)

Comment (nitinprabhu) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:56]: I think it's better to have method like `initialise` or `setUp` where you can club both `createDocker` and `startDocker` and get rid of below 
```
if (dockerContainer == null) {
             throw new RuntimeException(
                 "Cannot start container. Did you create one with "
                     + TestStorageHelper.class.getSimpleName()
                     + "::createDocker in a static context?"
             );
         }
```

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:56]: was random suggestion ^ ;). with if statement.

plus, it's generic class. helper. setup sounds more test related. i'll rename `initialiseStorage` ;)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+
+import static uk.gov.hmcts.reform.bulkscanprocessor.helper.DirectoryZipper.zipDir;
+
+public class TestStorageHelper {
+
+    private static TestStorageHelper INSTANCE;
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:18]: why can't we use proper names :) this is so generic

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        dockerContainer.start();
+    }
+
+    private static void initializeStorage() {
+        try {
+            cloudBlobClient = CloudStorageAccount
+                .parse("UseDevelopmentStorage=true")
+                .createCloudBlobClient();
+        } catch (InvalidKeyException | URISyntaxException exception) {
+            throw new RuntimeException("Unable to setup docker environment", exception);
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:52]: docker environment ? do we need try catch here ?

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:52]: try catch so to reduce exception signatures in test classes

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java (hunk 5 above)
Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:18]: i thought that's a normal approach when setting up instances of a class privately

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java (hunk 6 above)
Comment (nitinprabhu) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:52]: ok is this message valid `Unable to setup docker environment` ?

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/util/TestStorageHelper.java:52]: Now that method name is different, will replace with `unable to initialise storage` ;)

**Pull request merged**